File: web/dart/notify.dart
library scroll;

import 'dart:html';

bool renderNotify(Map<String, String> data) {
  DivElement el = querySelector('.notification');
  if (data['isOpen'] == false) {
     el.classes.add('none');
     return false;
  }
  el.classes.remove('none');
  el.querySelector('.message').text = data['message'];
  return true;
}

File: web/scripts/notify.ts
export function renderNotify(data: any): boolean {
  const element: HTMLElement | null = document.querySelector('.notification');

  if (!element) {
    return false;
  }

  if (!data.isOpen) {
    element.classList.add('none');
    return false;
  }

  element.classList.remove('none');

  const messageEl: HTMLElement | null = element.querySelector('.message');

  if (!messageEl) {
    return false;
  }

  messageEl.textContent = data.message;
  return true;
}
